dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt

```diff
@@ -86,15 +95,41 @@ internal class DataUploadRunnable(
 
     private fun isNetworkAvailable(): Boolean {
         val networkInfo = networkInfoProvider.getLatestNetworkInfo()
-        return networkInfo.connectivity != NetworkInfo.Connectivity.NETWORK_NOT_CONNECTED
+        val isOffline = networkInfo.connectivity == NetworkInfo.Connectivity.NETWORK_NOT_CONNECTED
+
+        if (isOffline) {
+            sendUploadQualityEvent(
```

Comment:
Minor: Not sure if this was expected but from `isNetworkAvailable` I expect that method just return some boolean value, and there is no side effects like calling`sendUploadQualityEvent`. I suggest call `sendUploadQualityEvent` in some separate pipeline or rename method to something like `resolveNetworkAvailability`.